docs/devGuide/design/serverSideRendering.md

```diff
@@ -53,10 +61,25 @@ The main motivation that we had for introducing SSR into MarkBind is to enhance
 
 SSR and Client-side Hydration are 2 concepts that go hand-in-hand. Essentially, once we produce the static HTML via SSR and send it over to the client-side, Vue on the client-side will execute what is known as Client-side Hydration on the static HTML.
 
-During the hydration process, Vue essentially `diff` your SSR HTML markup against the virtual DOM generated by the render function on the client-side. If any difference is found, meaning that the application that we have on the client-side (the virtual DOM tree) differs from the SSR HTML mark-up that we send to the client, Vue will reject the SSR HTML output, bail Client-side Hydration, and execute full CSR.
+During the hydration process, Vue essentially compares your SSR HTML markup against the virtual DOM generated by the render function on the client-side. If any difference is found, meaning that the application that we have on the client-side (the virtual DOM tree) differs from the SSR HTML mark-up that we send to the client, Vue will reject the SSR HTML output, bail Client-side Hydration, and execute full CSR.
 
 This is known as "Hydration Issue" and it is one of the main challenges you will face with SSR in MarkBind. 
 
+<box type="warning" header="##### Development vs. Production Build">
+
+There are some significant differences in how Vue **reports and handles hydration issues** between development and production builds.
+
+!!**Development Build**!!
+
+In development build, when Vue reports hydration issue, it will also provide information about the **exact mismatching node(s)** that is/are causing the hydration issue and inform the user that full CSR will be executed. Typically, it is very difficult to resolve hydration issues without the detailed error logs provided by the development build. 
```

Comment:
The box seems a little large -- should we remove this paragraph (also the 'there are some...' paragraph)? It seems explained by the section itself already, but I might have missed something.

Reply:
@wxwxwxwx9 I think missed this one

Reply:
@ang-zeyu Oops, yup I missed that part. Thanks for the spot. Removed it!

Reply:
@wxwxwxwx9

Hmm I actually meant to remove the entire `Development build` paragraph at first actually so this box only highlights what happens in production. Its a minor nit with keeping the alert box small.

and avoiding repeating what happens in development vue builds which was illustrated in your earlier paragraph accurately.
```
During the hydration process, Vue essentially compares your SSR HTML markup against the virtual DOM generated by the render function on the client-side. If any difference is found, meaning that the application that we have on the client-side (the virtual DOM tree) differs from the SSR HTML mark-up that we send to the client, Vue will reject the SSR HTML output, bail Client-side Hydration, and execute full CSR.
```

>Development Build
>In development build, when Vue reports hydration issue, it will also provide information about the exact mismatching node(s) that is/are causing the hydration issue and inform the user that full CSR will be executed.

> In production build, when Vue reports hydration issue, it will only log a generic "Failed to execute 'appendChild' on 'Node': This node type does not support this method" error message and execute full CSR silently. We have even observed cases where the production build does not warn about existing hydration issues (silent failures).
>
> The above reasons are why MarkBind uses the development build by default when you serve it in development mode (-d option).

The comparison may add confusion as -- note the `Failed to execute 'appendChild' on 'Node': This node type does not support this method` message is non standard, it'll vary depending on the error resulting from hydration itself. e.g. Is it a missing node? A node with an attribute that shouldn't be there? An extra node? -- many possible causes. Such errors may also not appear on page start (e.g. surfacing only when you click on a button that does some dom manipulation).

We could simply state that the production build does not perform the hydration check at all. Any such errors resulting from hydration prompts a full CSR render again. And then highlight that its not wise to test hydration errors with production bulids.

Reply:
> Hmm I actually meant to remove the entire `Development build` paragraph at first actually so this box only highlights what happens in production. Its a minor nit with keeping the alert box small.

Got it! Misinterpreted it in the beginning 😓 

> The comparison may add confusion as -- note the `Failed to execute 'appendChild' on 'Node': This node type does not support this method` message is non standard, it'll vary depending on the error resulting from hydration itself. e.g. Is it a missing node? A node with an attribute that shouldn't be there? An extra node? -- many possible causes. Such errors may also not appear on page start (e.g. surfacing only when you click on a button that does some dom manipulation).
> 
> We could simply state that the production build does not perform the hydration check at all. Any such errors resulting from hydration prompts a full CSR render again. And then highlight that its not wise to test hydration errors with production bulids.

Thanks for the suggestion! I've made the changes already :-)